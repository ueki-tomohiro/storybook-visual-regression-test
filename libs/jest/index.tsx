import { composeStories, StoryFile } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import React from "react";

export const renderFromStories = (stories: StoryFile) => {
  const composed = composeStories(stories);
  const testCases = Object.values(composed).map((Story: any) => [Story.storyName, Story, Story.args]);
  test.each(testCases)("renders %s", (_storyName, Component: any, args) => {
    if (_storyName.match(/Fake/)) return;

    const tree = render(<Component {...(args || {})} />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
