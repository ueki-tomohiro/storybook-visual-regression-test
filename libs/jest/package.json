{
  "name": "@demo-libs/jest",
  "version": "0.0.1",
  "private": true,
  "description": "Demo Jest",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-typescript": "^7.21.4",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.1",
    "babel-jest": "^29.5.0",
    "babel-plugin-transform-import-meta": "^2.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0"
  },
  "engines": {
    "node": "^20",
    "yarn": ">=1"
  },
  "volta": {
    "node": "20.0.0",
    "yarn": "1.22.19"
  }
}
