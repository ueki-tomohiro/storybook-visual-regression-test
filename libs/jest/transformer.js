const babelJest = require("babel-jest");
const babelOptions = {
  presets: [
    [
      "@babel/env",
      {
        targets: {
          node: "current",
        },
        exclude: ["transform-regenerator"],
      },
    ],
    "@babel/preset-typescript",
    "@babel/react",
  ],
  plugins: ["babel-plugin-transform-import-meta"],
  babelrc: false,
};
module.exports = babelJest.createTransformer(babelOptions);
