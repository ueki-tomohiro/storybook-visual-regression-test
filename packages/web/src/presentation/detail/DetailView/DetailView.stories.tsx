import { getGetTodoMock } from "@demo/api/lib/demo/todo/todo.msw";

import { DetailView, DetailViewProps } from "./DetailView";

const defaultArgs: DetailViewProps = {
  todo: {
    todo_id: 86624,
    description: "Hybrid",
    end_date: "2022-09-03T15:52:59.000Z",
    completed: true,
  },
};

export default {
  title: "views/DetailView",
  component: DetailView,
};

export const Default = { args: defaultArgs };

export const Loading = { args: { isLoading: true } };

export const Fake = {
  args: {
    ...defaultArgs,
    todo: getGetTodoMock(),
  },
};
