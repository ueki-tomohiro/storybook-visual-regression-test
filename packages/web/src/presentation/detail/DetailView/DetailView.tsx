import { Todo } from "@demo/api/lib/demo/model";
import { LinkButton } from "@demo/ui/components/LinkButton";
import dayjs from "@demo-libs/dayjs";
import React from "react";

export type DetailViewProps = {
  todo?: Todo;
};

export const DetailView: React.FC<DetailViewProps> = ({ todo }) => {
  if (!todo) return null;

  return (
    <>
      <div className="mb-4 flex items-center gap-2">
        <div className="text-xl">{todo.description}</div>
        <div className="text-xs text-tertiary">{dayjs(todo.end_date).tz().format("YYYY-MM-DD HH:mm:ss")}</div>
      </div>
      <div className="mb-4 flex items-center">
        {todo.completed ? (
          <>
            <span className="material-symbols-outlined">done</span>完了済み
          </>
        ) : (
          <>
            <span className="material-symbols-outlined">label_important</span>未完了
          </>
        )}
      </div>
      <LinkButton href={`/todo/${todo.todo_id}/edit`} variant="secondary">
        編集
      </LinkButton>
    </>
  );
};
