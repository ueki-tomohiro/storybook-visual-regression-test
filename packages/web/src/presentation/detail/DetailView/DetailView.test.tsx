import { renderFromStories } from "@demo-libs/jest";

import * as stories from "./DetailView.stories";

describe(stories.default.title, () => {
  renderFromStories(stories);
});
