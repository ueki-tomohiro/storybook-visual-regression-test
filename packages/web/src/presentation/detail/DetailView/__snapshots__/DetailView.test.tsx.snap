// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`views/DetailView > views/DetailView > "Default" story > renders without crashing 1`] = `
<div>
  <div
    class="mb-4 flex items-center gap-2"
  >
    <div
      class="text-xl"
    >
      Hybrid
    </div>
    <div
      class="text-tertiary text-xs"
    >
      2022-09-03 15:52:59
    </div>
  </div>
  <div
    class="mb-4 flex items-center"
  >
    <span
      class="material-symbols-outlined"
    >
      done
    </span>
    完了済み
  </div>
  <a
    class="mx-auto max-w-3xl text-white bg-secondary rounded px-4 py-2"
    href="/todo/86624/edit"
  >
    編集
  </a>
</div>
`;

exports[`views/DetailView > views/DetailView > "Loading" story > renders without crashing 1`] = `<div />`;
